modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/query_builder_tab/SearchableTypesModal.js

```diff
@@ -56,177 +52,189 @@ function SearchableTypesModal({
-																}
-																onChange={_handleRowCheck(
-																	className
-																)}
-															/>
-														</ClayTable.Cell>
-
-														<ClayTable.Cell
-															expanded
-															headingTitle
-														>
-															{displayName}
-														</ClayTable.Cell>
-													</ClayTable.Row>
-												);
-											}
-										)}
-									</ClayTable.Body>
-								</ClayTable>
-							</ClayModal.Body>
-						</>
-					) : (
-						<ClayModal.Body>
-							<ClayEmptyState
-								description={Liferay.Language.get(
-									'an-error-has-occurred-and-we-were-unable-to-load-the-results'
+													/>
+												</ClayTable.Cell>
+
+												<ClayTable.Cell
+													expanded
+													headingTitle
+												>
+													{displayName}
+												</ClayTable.Cell>
+											</ClayTable.Row>
+										);
+									}
 								)}
-								imgSrc="/o/admin-theme/images/states/empty_state.gif"
-								title={Liferay.Language.get(
-									'no-items-were-found'
-								)}
-							>
-								<ClayButton
-									displayType="secondary"
-									onClick={onFetchSearchableTypes}
-								>
-									{Liferay.Language.get('refresh')}
-								</ClayButton>
-							</ClayEmptyState>
-						</ClayModal.Body>
-					)}
-
-					<ClayModal.Footer
-						last={
-							<ClayButton.Group spaced>
-								<ClayButton
-									displayType="secondary"
-									onClick={onClose}
-								>
-									{Liferay.Language.get('cancel')}
-								</ClayButton>
-
-								<ClayButton onClick={_handleModalDone}>
-									{Liferay.Language.get('done')}
-								</ClayButton>
-							</ClayButton.Group>
-						}
-					/>
-				</ClayModal>
+							</ClayTable.Body>
+						</ClayTable>
+					</ClayModal.Body>
+				</>
+			) : (
+				<ClayModal.Body>
+					<ClayEmptyState
+						description={Liferay.Language.get(
+							'an-error-has-occurred-and-we-were-unable-to-load-the-results'
+						)}
+						imgSrc="/o/admin-theme/images/states/empty_state.gif"
+						title={Liferay.Language.get('no-items-were-found')}
+					>
+						<ClayButton
+							displayType="secondary"
+							onClick={onFetchSearchableTypes}
+						>
+							{Liferay.Language.get('refresh')}
+						</ClayButton>
+					</ClayEmptyState>
+				</ClayModal.Body>
+			)}
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton displayType="secondary" onClick={onClose}>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+
+						<ClayButton onClick={_handleModalDone}>
+							{Liferay.Language.get('done')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</ClayModal>
+	);
+}
+
+export default function ({
+	children,
+	onFetchSearchableTypes,
+	onFrameworkConfigChange,
+	searchableTypes,
+	selectedTypes,
+}) {
+	const [visible, setVisible] = useState(false);
+
+	const {observer, onClose} = useModal({
+		onClose: () => setVisible(false),
+	});
```

Comment:
Think you can reduce this to:

```javascript
const {observer, onOpenChange, open} = useModal();
```

See https://clayui.com/docs/components/modal.html